SafeguardPasswords/Src/Modules/Request/Request.pqm

```diff
@@ -0,0 +1,147 @@
+let  
+    ValidateByStatusCode = Extension.ImportFunction("ValidateByStatusCode", "ResponseHandler.pqm"),
+    InfoLog = Extension.ImportFunction("InfoLog", "CommonLogger.pqm"),
+    ErrorCodes = Extension.ImportFunction("ErrorCodes", "RequestErrors.pqm"),
+    GetHost = Extension.ImportFunction("GetSppHost","UrlBuilder.pqm"),
+    Constants = Extension.ImportModule("Constants.pqm")[SppConstants],
+    IsMocked = Extension.ImportFunction("IsMocked", "CommonUtils.pqm"),
+
+
+    Request.GetSppResponseBase = (host as text, options as record, optional logResponse as logical, optional mocks as record) =>
+        let
+
+            // IsRetry causes Web.Contents to not use cached API response. 
+            // See https://learn.microsoft.com/en-us/powerquery-m/web-contents for details
+            response = if IsMocked(mocks, "Response")
+                        then mocks[Response]
+                        else Web.Contents(host, options & [ManualStatusHandling = {400,406}, IsRetry = true]),
+
+            json = if logResponse = true 
+                    then InfoLog("Response from SPP: ", Json.Document(response))
+                    else Json.Document(response),
+
+            extendedMeta = Value.Metadata(response) & [RequestUrl = host] & [RelativePath = options[RelativePath]],// & [Query = options[Query]],
```

Comment:
Yes, missed that I commented that out. 

The API Query is passed down to web.Contents by means of options; I excluded it from the extendedMeta because, if I recall correctly, at the time I was trying to decide whether or not I wanted it repeated in logging. I'll uncomment it.